### PR TITLE
ENT-5012: Refactor Metric ID to the rhMarketplaceMetricId

### PR DESCRIPTION
**Pull request by ksynvrit**
This PR is to address sub task: https://issues.redhat.com/browse/ENT-5012 

Rename metric id to rhMarketplaceMetricId in tag_profile.yaml.
Update implementing Unit tests

### CODE REVIEW

File: src/main/java/org/candlepin/subscriptions/metering/MeteringEventFactory.java [hunk 1]
```diff
@@ -127,9 +127,9 @@ public static void updateMetricEvent(
...
   }
 
-  public static String getEventType(String metricId) {
-    return StringUtils.hasText(metricId)
-        ? String.format("%s_%s", EVENT_TYPE, metricId)
+  public static String getEventType(String rhMarketplaceMetricId) {
+    return StringUtils.hasText(rhMarketplaceMetricId)
+        ? String.format("%s_%s", EVENT_TYPE, rhMarketplaceMetricId)
         : EVENT_TYPE;
   }
```
Comment (kahowell) [inline src/main/java/org/candlepin/subscriptions/metering/MeteringEventFactory.java:135]: Actually for MeteringEventFactory, we should keep it as `metricId`. Unfortunately, we overloaded `metricId` at this point. Sometimes its a canonical identifier for a metric; sometimes its the RHM-specific identifier. Please revert changes in this file.

File: src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMeteringController.java [hunk 2]
```diff
@@ -260,7 +260,7 @@ private Event createOrUpdateEvent(
     MeteringEventFactory.updateMetricEvent(
         event,
         account,
-        metricId,
+        rhMarketplaceMetricId,
```
Comment (kahowell) [inline src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMeteringController.java:263]: Similar to previous comment, please revert changes in this file.

File: src/test/java/org/candlepin/subscriptions/metering/service/prometheus/promql/QueryBuilderTest.java [hunk 3]
```diff
@@ -94,13 +100,13 @@ void supportsNestedExpressions() {
...
-                .metricId(metricId)
+                .rhMarketplaceMetricId(metricId)
                 .queryParams(queryParams)
                 .build());
     queryDesc.addRuntimeVar("account", account);
 
     QueryBuilder builder = new QueryBuilder(props);
     String query = builder.build(queryDesc);
-    assertEquals(String.format("Account: %s Metric ID: %s", account, metricId), query);
+    assertEquals(String.format("Account: %s RhMarketplaceMetric ID: %s", account, metricId), query);
```
Comment (kahowell) [inline src/test/java/org/candlepin/subscriptions/metering/service/prometheus/promql/QueryBuilderTest.java:110]: Please revert changes in this file.

**Review by kahowell — CHANGES_REQUESTED** [review]

File: src/main/java/org/candlepin/subscriptions/metering/MeteringEventFactory.java (hunk 1 above)
Comment (ksynvrit) [reply src/main/java/org/candlepin/subscriptions/metering/MeteringEventFactory.java:135]: Alright will do

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
30d30b7d-116c-485d-bda2-ce4407ae2b68
e3e2d07a-db98-4278-83e6-e35e499904e9

**Pull request merged**